net/inet_pton.c

```diff
@@ -1,6 +1,146 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libphoenix
+ *
+ * net/inet_pton.c
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Maciej Purski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
 #include <arpa/inet.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include <string.h>
 
-int inet_pton(int af, const char *restrict src, void *restrict dst)
+/* Accept only addresses in dotted-decimal format ddd.ddd.ddd.ddd */
+static int inet_pton4(const char *src, struct in_addr *dst)
 {
-	return 0;
+	in_addr_t addr4 = 0;
+	unsigned long val;
+	int i;
+	char *endptr;
+
+	for (i = 0; i < 4; i++) {
+		if (!isdigit(*src))
+			return 0;
+
+		val = strtoul(src, &endptr, 0);
+		if (endptr == src || val > 255)
+			return 0;
+		if ((*endptr != '.' && i < 3) ||
+			(*endptr != '\0' && i == 3))
+			return 0;
+
+		src = endptr + 1;
+		addr4 = addr4 << 8 | (val & 0xFFFF);
```

Comment:
```suggestion
		addr4 = addr4 << 8 | val ;
```
`val` is `unsigned int` and `<= 255`